### DISCUSSION TITLE
Localization Bundles

### DISCUSSION THREAD

**Discussion by Kitosight** [microsoft/vscode-discussions#505] [Extension Development QnA] [answered]
Can I use `.jsonc` (JSON with comments) for localization bundle files?

**Comment by alefragnani** [comment, marked as answer]
Hi @NightNutSky ,

What do you mean by _localization bundle files_? I'm asking because I see at least 3 different bundles

### Translation

Where you translate the contents from `package.json` like commands, settings, etc (`package.nls.json`)

It is not supported. If you rename the file to `package.nls.jsonc` the translation does not take effect. And if instead, you simply add comments inside the `.json` file, the translation will work (at debugging time) but you will receive an error while packaging (`vsce ls`)
```
 ERROR  Error parsing JSON manifest translations file: /Volumes/myuser/sources/vscode-copy-word/package.nls.json
```

### Localization

Where you translate resources inside the extension source, like messages (`l10n/bundle.l10n.en.json`). A new and improved API was released a couple of months ago.

It is not supported. No matter if you rename the file to `bundle.l10n.en.jsonc` or simply add comments to the `.json` file, the translation does not take effect. While packaging, on the other hand, does not display any error.

### Language Packs

Where you translate the VS Code itself to another language (most of the packs comes from MS - https://github.com/microsoft/vscode-loc).

It's been a long time I don't do that, but based on a quick search, it seems it does not work either.

## But, you still can do it

The good part is that both **Translation** and **Localization**, supports a `comment` entry in the `json` file, so you can give some context to the translator.  

Here are some references in Translation (https://github.com/microsoft/vscode/blob/5271145f3ff283fde3dbb58d9a18b0ee3bfff478/extensions/git/package.nls.json#L198) and Localization (https://github.com/microsoft/vscode-l10n/tree/main/l10n-dev#exporting-strings).

Hope this helps

**Reply by Kitosight** [reply]
Okay, so..

> `bundle.l10n.json`
```json
{
    "Some string for localization": {
        "message": "Some string for localization",
        "comment": ["Comment here"]
    }
}
```

> `bundle.l10n.ru.json`
```json
{
    "Some string for localization": "Какая-то строка для локализации"
}
```

Is it right? (I mean, must I add `/This is a comment?` in a string name or is it added there just to indicate that this string has a comment?) Please explain better if I understood it wrongly.

**Reply by alefragnani** [reply]
Yes, it is right. The `comment` entry doesn't need to be replicated in translated bundles

As commented in https://github.com/microsoft/vscode-l10n/tree/main/l10n-dev#exporting-strings

> You don't need the comments in the localized bundles since the comments are only useful for translators translating the original bundle.

**Reply by Kitosight** [reply]
Thanks.
